dependencies.py

```diff
@@ -205,26 +234,59 @@ def _ensure_prerequisite_is_installed(prerequisite="pip"):
         raise RuntimeError(msg)
 
 
-def _dependencies_target_dir(our_dir=OUR_DIR):
-    """Return python dir inside our profile
+def _dependencies_target_dir(*, our_dir=OUR_DIR, create=False) -> Path:
+    """Returns (and creates) the desired deps folder
 
-    Return two dirs up if we're inside the plugins dir. If not, we have to
-    import from qgis (which we don't really want in this file) and ask for our
-    profile dir.
+    This is the 'deps' subdirectory of the plugin home folder
 
+    """
+    target_dir = our_dir / 'deps'
+    if not target_dir.exists() and create:
+        print(f'Creating target dir {target_dir}')
+        target_dir.mkdir()
+
+    return target_dir
+
+
+def _prev_dependencies_target_dir(our_dir=OUR_DIR) -> Path:
+    """Return python dir inside our profile
+
+    Return two dirs up if we're inside the plugins dir. This was the
+    previous installation folder of the dependencies.
     """
     if "plugins" in str(our_dir).lower():
-        # Looks like we're in the plugin dir. Return ../..
         return OUR_DIR.parent.parent
-    # We're somewhere outside of the plugin directory. Perhaps a symlink?
-    # Perhaps a development setup? We're forced to import qgis and ask for our
-    # profile directory, something we'd rather not do at this stage. But ok.
-    print("We're not in our plugins directory: %s" % our_dir)
-    from qgis.core import QgsApplication
 
-    python_dir = Path(QgsApplication.qgisSettingsDirPath()) / "python"
-    print("We've asked qgis for our python directory: %s" % python_dir)
-    return python_dir
+
+def _remove_old_distributions(dependencies, path) :
+    """Remove old distributions of dependencies
+
+    In previous version of the Toolbox, depencencies were
+    stored in the users 'python' folder. This caused
+    versioning conflicts (as these dependencies were
+    not removed when the plugin was uninstalled).
+
+    Removes all folders and files that contain the
+    dependency name or package name
+    """
+    succeeded = True
+    files_to_remove = [node for node in os.listdir(str(path)) for dependency in dependencies if (dependency.package in node or dependency.name in node)]
+
+    for f in files_to_remove:
+        dep_path = str(path / f)
+
+        try:
+            if os.path.exists(dep_path):
+                if os.path.isfile(dep_path):
+                    print(f'Deleting file {f} from {path}')
+                    os.remove(dep_path)
+                else:
+                    print(f'Deleting folder {f} from {path}')
+                    shutil.rmtree(dep_path)
+        except PermissionError:
+            succeeded = False
```

Comment:
Ik zou nog even wat printen. En een `as e` bij de PermissionError toevoegen en die meenemen in de print.